mapcache.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2019 Ricardo Leite. All rights reserved.
+ * Licenced under the MIT licence. See COPYING file in the project root for
+ * details.
+ */
+
+#ifndef __MAPCACHE_H_
+#define __MAPCACHE_H_
+
+#include <sys/mman.h>
+#include "log.h"
+#include "size_classes.h"
+
+#define MAPCACHE_SIZE 64
+
+struct MapCacheBin {
+private:
+    char* _block = nullptr;
+    uint32_t _blockNum = 0;
+
+public:
+    char *Alloc();
+    void Flush();
+};
+
+inline char* MapCacheBin::Alloc()
+{
+	if (_blockNum == 0) {
+		_block = (char*)mmap(nullptr, SB_SIZE * MAPCACHE_SIZE, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+		ASSERT(_block != MAP_FAILED);
+		_blockNum = MAPCACHE_SIZE;
+	}
```

Comment:
some indentation issues here, can see it on the diff quite clearly

Reply:
Fixed. Formatted all files according to the .clang-format file